src/librbd/cache/pwl/InitRequest.cc

```diff
@@ -178,12 +178,6 @@ void InitRequest<I>::handle_set_feature_bit(int r) {
     shutdown_image_cache();
   }
 
-  if (m_image_ctx.discard_granularity_bytes) {
```

Comment:
@majianpeng is it true to assume that we would not need discard_granualarity_bytes reset when an existing image is init with cache?

Reply:
Yes, apply_metadata call two times. One time before init and other time after init. So in init function reset don't affect.